code-challenges.test.js

```diff
@@ -14,38 +14,122 @@
+    { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+    { name: "arthur dent", occupation: "a radio employee" }
+  ]
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized", () => {
+    expect(capPeople(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
+//4. create a function capPeople take in an array
+//5. map trough every index and .split(" ") then .map() and replace value[0] to .toUpperCase()
+//6. .join(" ") 
+//7. return value of fullname and description in a sentence `${fullname} is ${desc}.`
+//8. run yarn jest and got green
+
+const capPeople = (array) => {
+return array.map(nameDesc => {
+  let name = nameDesc.name.split(" ").map(value => value.replace(value[0], value[0].toUpperCase()))
+  let fullname = name.join(" ")
+  let desc = nameDesc.occupation
+  let nameTitle = `${fullname} is ${desc}.`
+  return nameTitle
+})
+}
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+// const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+// // Expected output: [ 2, 0, -1, 0 ]
+// const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+// // Expected output: [ 2, 1, -1 ]
 
+//1. create a test numberOnly and take in a mixed data array and returns an arrray of only the remainders of the numbers when divided by 3
+//2. expect 23, "Heyyyy!", 45, -10, 0, "Yo", false] toEqual [ 2, 0, -1, 0 ]
+//3. run yarn jest and got red
+
+describe("numberOnly", () => {
+  const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+  const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+  it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+    expect(numberOnly(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(numberOnly(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
+//4. create a function numberOnly and take an array a parameter
+//5. .filter() and take only typeof number the .map() to run through every index in the array 
+//6. return value % 3
+//7. run yarn jest and got green
+
+//REFACTOR
+const numberOnly = (array) => {
+  return array.filter(value => typeof value === "number").map(x => { return x % 3 }
+)}
+
+//OG Code
+// const numberOnly = (array) => {
+//   let newArr = array.filter(value => typeof value === "number")
+//   return newArr.map(x => { return x % 3 }
+// )}
+
```

Comment:
Good job refactoring. I like how you work it the way you understood then went back and consolidated.